qiskit_quantuminspire/qi_jobs.py

```diff
@@ -35,15 +59,92 @@ def __init__(
         super().__init__(backend, job_id, **kwargs)
         self._job_ids: List[str] = []
         self._run_input = run_input
+        self.program_name = "Program & Algorithm created by SDK"
 
-    def submit(self) -> None:
+    async def submit(self) -> None:
         """Submit the (batch)job to the quantum inspire backend.
 
         Use compute-api-client to call the cjm endpoints in the correct order, to submit the jobs.
         """
-        for i in range(1, 3):
-            self._job_ids.append(str(i))
-        self.job_id = "999"  # ID of the submitted batch-job
+        options = self.backend().options
+        configuration = config()
+        # call create algorithm
+        async with ApiClient(configuration) as api_client:
+
+            # always create a list
+            circuits: List[QuantumCircuit] = []
+            if isinstance(self._run_input, QuantumCircuit):
+                circuits.append(self._run_input)
+            else:
+                circuits = self._run_input
+
+            project = await self._create_project(api_client, configuration.auth_settings().team_member_id)
+            batch_job = await self._create_batch_job(api_client, backend_type_id=self.backend().id)
+
+            # iterate over the circuits
+            for circuit in circuits:
```

Comment:
Should be done in parallel using asyncio.gather().

Reply:
Done.